SU2_CFD/src/iteration/CFluidIteration.cpp

```diff
@@ -107,26 +109,24 @@ void CFluidIteration::Iterate(COutput* output, CIntegration**** integration, CGe
 
   /*--- If the flow integration is not fully coupled, run the various single grid integrations. ---*/
 
-  if ((config[val_iZone]->GetKind_Solver() == MAIN_SOLVER::RANS || config[val_iZone]->GetKind_Solver() == MAIN_SOLVER::DISC_ADJ_RANS ||
-       config[val_iZone]->GetKind_Solver() == MAIN_SOLVER::INC_RANS || config[val_iZone]->GetKind_Solver() == MAIN_SOLVER::DISC_ADJ_INC_RANS) &&
-      !frozen_visc) {
+  if ((main_solver == MAIN_SOLVER::RANS) && !frozen_visc) {
     /*--- Solve the turbulence model ---*/
 
-    config[val_iZone]->SetGlobalParam(MAIN_SOLVER::RANS, RUNTIME_TURB_SYS);
+    config[val_iZone]->SetGlobalParam(main_solver, RUNTIME_TURB_SYS);
     integration[val_iZone][val_iInst][TURB_SOL]->SingleGrid_Iteration(geometry, solver, numerics, config,
                                                                       RUNTIME_TURB_SYS, val_iZone, val_iInst);
 
     /*--- Solve transition model ---*/
 
     if (config[val_iZone]->GetKind_Trans_Model() == TURB_TRANS_MODEL::LM) {
-      config[val_iZone]->SetGlobalParam(MAIN_SOLVER::RANS, RUNTIME_TRANS_SYS);
+      config[val_iZone]->SetGlobalParam(main_solver, RUNTIME_TRANS_SYS);
       integration[val_iZone][val_iInst][TRANS_SOL]->SingleGrid_Iteration(geometry, solver, numerics, config,
                                                                          RUNTIME_TRANS_SYS, val_iZone, val_iInst);
     }
   }
 
   if (config[val_iZone]->GetKind_Species_Model() != SPECIES_MODEL::NONE){
-    config[val_iZone]->SetGlobalParam(config[val_iZone]->GetKind_Solver(), RUNTIME_SPECIES_SYS);
+    config[val_iZone]->SetGlobalParam(main_solver, RUNTIME_SPECIES_SYS);
```

Comment:
This should be why it was not working, turbulence did not suffer from the same problem.